src/scancode/cli.py

```diff
@@ -362,6 +363,9 @@ def save_results(scanned_files, format, input, output_file):
     """
     Save results to file or screen.
     """
+    if output_file != sys.stdout:
+        fileutils.create_dir(os.path.dirname(output_file.name))
```

Comment:
You may want to consider the case where the path is does not contain any directory but is just a plain file. In case, the parent directory name may be empty. So may be something like that may help:
```
        parent_dir = os.path.dirname(output_file.name)
        if parent_dir:
            fileutils.create_dir(parent_dir)
```
There is may be another case to consider for userdir expansion (e.g. if some `~/some/path/to/file` was used). In that case  then using this may be better:
```
from os.path import expanduser
from os.path import abspath
```
```
        parent_dir = os.path.dirname(output_file.name)
        if parent_dir:
            fileutils.create_dir(abspath(expanduser(parent_dir)))
```